app/controllers/promotions_controller.rb

```diff
@@ -1,6 +1,9 @@
+# PRO: Clean controller, nice!
 class PromotionsController < ApplicationController
     def current_promotion
         final_price = Promotions::GeneralPriceService.get_price_from_current_promotion(price_param)
+        # CON: Missing the discounted price for each individual item, as
```

Comment:
it's in the promotion service, so that the controller will not be the big size

Reply:
hey @laranjeira-bonita , yep it makes sense to use a service here to avoid complexity on the controller. However, my point here is that we're missing a return of the discounted price for each individual item. For example, taking the challenge description:

_Your mission is to write an API to calculate the price of any conceivable shopping cart (containing cereal boxes from multiple collections), **returning the discounted price for each individual item** and the final discounted price for the entire cart._

We expected to have the final price and the final price for each individual product as well, taking for example the provided JSON input, a valid output from the API would be something around these lines:

```JSON
{
  "cart": {
    "total_price": "130.38"
    "reference": "2d832fe0-6c96-4515-9be7-4c00983539c1",
    "lineItems": [
      { "name": "Peanut Butter", "price": "39.0", "discountedPrice": "35.10", "collection": "BEST-SELLERS" },
      { "name": "Banana Cake", "price": "34.99", "discountedPrice": "31.49", "collection": "DEFAULT" },
      { "name": "Cocoa", "price": "34.99", "discountedPrice": "34.99", "collection": "KETO" },
      { "name": "Fruity", "price": "32", "discountedPrice": "28.8", "collection": "DEFAULT" }
    ]
  }
}
```

I've added a `discountedPrice` for each item on the response. Does that make sense?